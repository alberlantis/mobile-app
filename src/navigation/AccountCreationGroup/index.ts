export { default } from "./AccountCreationGroup";
export type { AccountCreationParamList } from "./AccountCreationGroup";
