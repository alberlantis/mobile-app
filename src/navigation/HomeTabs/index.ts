export { default } from "./HomeTabs";
export type { HomeTabsParamList } from "./HomeTabs";
