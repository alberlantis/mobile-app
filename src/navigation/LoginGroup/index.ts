export { default } from "./LoginGroup";
export type { LoginGroupParamList } from "./LoginGroup";
