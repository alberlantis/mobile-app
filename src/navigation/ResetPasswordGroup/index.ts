export { default } from "./ResetPasswordGroup";
export type { ResetPasswordGroupParamList } from "./ResetPasswordGroup";
