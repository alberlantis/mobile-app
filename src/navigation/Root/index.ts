export { default } from "./Root";
export type {
  RootScreenProps,
  RootNavigationProps,
  RootRouteProps,
  HomeTabsParams,
} from "./Root";
