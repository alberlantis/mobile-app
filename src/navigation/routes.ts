export const SCREENS = {
  HOME_TABS: "HomeTabs",
  HOME: "Home",
  NOTIFICATIONS: "Notifications",
  ONBOARDING: "Onboarding",
  POSTING: "Posting",
  PROFILE_HOME: "ProfileHome",
  SPLASH: "Splash",
  SIGN_UP: "SignUp",
  NOSTR_UP: "NostrUp",
  LOGIN: "Login",
  NOSTR_IN: "NostrIn",
  INTERESTS: "Interests",
  COMPLETE_PROFILE: "CompleteProfile",
  RECOVERY_EMAIL: "RecoveryEmail",
  RESET_PASSWORD: "ResetPassword",
} as const;
