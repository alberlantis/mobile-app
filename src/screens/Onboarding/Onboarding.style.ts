import { StyleSheet } from "react-native";
import { normalizeSize } from "src/theme";

export default StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    paddingHorizontal: normalizeSize(32),
  },
  emailButton: {
    marginBottom: normalizeSize(16),
    marginTop: normalizeSize(32),
  },
  nostrButton: {
    marginBottom: normalizeSize(20),
  },
});
