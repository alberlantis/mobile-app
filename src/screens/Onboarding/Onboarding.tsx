import React from "react";
import { Text } from "react-native";

import { SCREENS } from "navigation/routes";
import { VersionLabel } from "shared/components";
import type { RootScreenProps } from "navigation/Root";

const Onboarding = ({ navigation }: RootScreenProps<"Onboarding">) => {
  return (
    <>
      <Text onPress={() => navigation.navigate(SCREENS.HOME_TABS)}>
        Onboarding Screen
      </Text>
      <VersionLabel />
    </>
  );
};

export default Onboarding;
