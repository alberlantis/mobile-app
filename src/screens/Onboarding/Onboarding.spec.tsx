import { fireEvent, render, screen } from "@testing-library/react-native";

import { SCREENS } from "src/navigation/routes";
import Onboarding from "./Onboarding";

jest.mock("src/shared/components", () => {
  const { Text } =
    jest.requireActual<typeof import("react-native")>("react-native");
  return {
    DefaultBackground: ({ children }: { children: React.ReactNode }) => (
      <>{children}</>
    ),
    Button: ({ text, onPress }: { text: string; onPress(): void }) => (
      <Text onPress={onPress}>{text}</Text>
    ),
    InteractiveText: ({
      text,
      prefix,
      onPress,
    }: {
      text: string;
      prefix: string;
      onPress(): void;
    }) => (
      <Text onPress={onPress}>
        {prefix}
        {text}
      </Text>
    ),
    LogoTitle: ({ title }: { title: string }) => <Text>{title}</Text>,
  };
});

const mockNavigation: any = {
  navigate: jest.fn(),
};
const mockRoute: any = {
  name: "Onboarding",
};

describe("Onboarding", () => {
  beforeEach(() => {
    render(<Onboarding navigation={mockNavigation} route={mockRoute} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should match snapshot", () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it("should render logo title", () => {
    expect(screen.getByText("Sign Up")).toBeOnTheScreen();
  });

  it("should render Using Email button", () => {
    expect(screen.getByText("Using Email")).toBeOnTheScreen();
  });

  it("should navigate to Sign Up when tapped", () => {
    const button = screen.getByText("Using Email");
    fireEvent.press(button);
    expect(mockNavigation.navigate).toHaveBeenCalledTimes(1);
    expect(mockNavigation.navigate).toHaveBeenCalledWith(SCREENS.SIGN_UP);
  });

  it("should render nsec button", () => {
    expect(screen.getByText("Using Nostr")).toBeOnTheScreen();
  });

  it("should navigate to NostrUp when tapped", () => {
    const button = screen.getByText("Using Nostr");
    fireEvent.press(button);
    expect(mockNavigation.navigate).toHaveBeenCalledTimes(1);
    expect(mockNavigation.navigate).toHaveBeenCalledWith(SCREENS.NOSTR_UP);
  });

  it("should render sign in text", () => {
    expect(
      screen.getByText("Already have an account?Sign in"),
    ).toBeOnTheScreen();
  });

  it("should navigate to Login when tapped", () => {
    const button = screen.getByText("Already have an account?Sign in");
    fireEvent.press(button);
    expect(mockNavigation.navigate).toHaveBeenCalledTimes(1);
    expect(mockNavigation.navigate).toHaveBeenCalledWith(SCREENS.LOGIN);
  });
});
