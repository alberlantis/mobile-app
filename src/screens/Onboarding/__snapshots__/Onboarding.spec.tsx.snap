// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding should match snapshot 1`] = `
[
  <Text>
    Sign Up
  </Text>,
  <Text
    onPress={[Function]}
  >
    Using Email
  </Text>,
  <Text
    onPress={[Function]}
  >
    Using Nostr
  </Text>,
  <Text
    onPress={[Function]}
  >
    Already have an account?
    Sign in
  </Text>,
]
`;
